src/ext/php7/engine_api.c

```diff
@@ -54,12 +54,14 @@ ZEND_RESULT_CODE ddtrace_call_method(zend_object *obj, zend_class_entry *ce, zen
         .retval = retval,
         .params = argv,
         .object = obj,
-        /* Must be 0 to allow for by-ref args
-         * BUT if you use by-ref args, you need to free the ref if it gets separated!
-         */
-        .no_separation = 0,
         .param_count = argc,
     };
+#if PHP_VERSION_ID < 80000
+    /* Must be 0 to allow for by-ref args
+     * BUT if you use by-ref args, you need to free the ref if it gets separated!
+     */
+    fci.no_separation = 0;
```

Comment:
Removal of `no_separation` means what for pass-by-ref args?

Reply:
Good question. I'm not sure the impact this will have at the moment. But according to the [upgrading doc](https://github.com/php/php-src/blob/3e33e1e/UPGRADING.INTERNALS#L144-L148):

> The zend_fcall_info no_separation flag has been removed, and separation is never allowed. If you wish to pass (or allow passing) arguments by reference, explicitly create those arguments as references using ZEND_MAKE_REF.

So this will require more investigation.

Reply:
So it looks like we might need to add `ZEND_MAKE_REF` to a few places to ensure this works as expected, but since none of the pass-by-reference tests failed, I'll need to dig a bit deeper. Would you be OK if we addressed this in the followup PR?

Reply:
IIRC the curl helper functions take an array by reference that might get separated. So perhaps dig into that specific case a little bit? As long as we understand what's going on we can delay to another PR, but I don't want to land something in master that is pretty suspect like this.

Reply:
I did a bit more digging and found that we have more blockers to do a proper investigation on the impact of this. The only place `ddtrace_call_{method|function}` is called with args is isolated to the curl handler which doesn't run on PHP 8 yet because it [needs to be upgraded](https://github.com/DataDog/dd-trace-php/blob/ba542dd/src/ext/php7/handlers_curl.c#L297) to use `CurlHandler` instead of a resource. Then we'll need to validate that the args for `GlobalTracer::get()->inject()` are handled properly.

Since this is blocked, are you OK with this landing in master now that we have a more specific plan of action to investigate this further?

Reply:
Sounds good to me.